### ISSUE TITLE
"This is my code, and after running it, it outputs NAN, which is confusing." 

### ISSUE THREAD

**Issue by zwfengineer** [pardeike/Harmony#627]
```
public class Class1
    {
        private  float  SeverityInit=0.0f;
        public   float Severity
        {
            get {
                return this.SeverityInit;
            }
            set {
                Console.WriteLine(value);
                this.SeverityInit = value;
            }
        }
    }
    public static class MyPatcher
    {
        // make sure DoPatching() is called at start either by
        // the mod loader or by your injector

        public static void DoPatching()
        {
            var harmony = new Harmony("com.example.patch");
            harmony.PatchAll();
        }
    }

    [HarmonyPatch(typeof(Class1), "Severity", MethodType.Getter)]

    class PatchSetSeverity
    {
        static bool Prefix(Class1 __instance)
        {
            Console.WriteLine("GetPrefix");
            return true;
        }
        static void  Postfix(ref float __result)
        {
            Console.WriteLine("GetPostfix::{0}", __result);
            __result += 5;
        }
    }
    internal class Program
    {
        static void Main(string[] args)
        {
            MyPatcher.DoPatching();

            Class1 c1 = new Class1();
            //c1.Severity = 10.0f;
            Console.WriteLine("{0}", c1.Severity);


            Console.ReadKey();

        }
    }

```
**Describe the bug**

**To Reproduce**
1. The original method and its signature and class
2. The patch code
3. The output of the Harmony debug log
4. The stacktrace or other errors

**Expected behavior**

**Screenshots / Code**
If applicable, add material to help explain your problem.

**Runtime environment (please complete the following information):**
 - OS: [e.g. Windows 10, 64bit]
 - .NET version [e.g. 3.5]
 - Harmony version [e.g. 1.2.0.1]
- Name of game or host application [e.g. RimWorld]

**Additional context**

**Comment by pardeike** [comment]
I copied your code into this online project and it works just fine:
https://replit.com/@pardeike/HarmonyTest#main.cs

Issues on GitHub are for bugs not support. Please use the official Discord server which gives you more help than here.

**Comment by zwfengineer** [comment]
> I copied your code into this online project and it works just fine: https://replit.com/@pardeike/HarmonyTest#main.cs
> 
> Issues on GitHub are for bugs not support. Please use the official Discord server which gives you more help than here.


I found the cause of the problem. I was using the default "Any CPU" mode in Visual Studio. After I adjusted the CPU mode, I encountered the following results: it returns NAN under the x86 architecture, works fine under x64, and is not supported on ARM64 (definitely not supported...). It would be best to mention this in the documentation. I never would have thought it was an issue with the CPU architecture.
